include/picongpu/particles/MoveParticle.hpp

```diff
@@ -34,10 +34,12 @@ namespace picongpu
          * in case the new position is outside of the cell we adjust
          * the cell index and the multiMask of the particle.
          *
+         * we assume that CFL condition is fulfilled
```

Comment:
Not sure if CFL (Courant–Friedrichs–Lewy condition) is correct here @PrometheusPi @steindev could you please have a look.

Reply:
well its is CFL adjacent, but not strictly CFL.

We guarantee that a particle may never move more than one cell length separately from the CFL condition as described here.

https://picongpu.readthedocs.io/en/latest/models/AOFDTD.html#substepping

But I am not exactly sure how that will interact with the differrent CFL condiions of AOFDTD.

Reply:
Well, I'm happy to revise this into something more explicit like "We guarantee that a particle may never move more than one cell length". Just think that there should be something in there stating that (if not checking that in code) -- at best with a hint about the physical/numerical explanation.

Reply:
IMO we can safely write it based on the PMacc restriction that a particle is not allowed to move more than one cell.
This restriction is not physically motivated it is simply an implementation restriction.
Additionally, there are restrictions from PIConGPU. The current deposition schema also restricts the largest move to one cel and field solvers have higher restrictions depending on the solver type.